runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -5714,8 +5707,13 @@ void get_buff_thread_info_destroy(void* p) {
       global_info = global_info->next;
     }
   }
-
   rwlock_unlock(&get_buff_global_info.lock);
+
+  // flush any pending ops
+  spinlock_lock(&info->lock);
```

Comment:
We don't need to lock the entry since it's not in the list and nobody else could be looking at it, do we?  (Though it doesn't hurt anything functionally.)

Reply:
I had the same thought, but decided to keep it out of paranoia. For sure we don't need the mutual exclusion part of the lock, but we may need the memory fence (though I'm not positive about that)

Reply:
> ... we may need the memory fence ...

Possibly, if this thread frees the mem and then another thread grabs it for the same purpose, yeah.